code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -40,102 +52,276 @@
+	desc = "An experimental suit of armor with a reactive sensor array rigged to flame jet gloves. Pyromania with style AND percision, just in case you don't want to burn down your entire department like the first model did."
+	reactive_icon = "reactive_flame"
+	icon_state = "reactiveoff_flame"
+	inhand_icon_state = "reactiveoff_flame"
+	cooldown_message = "<span class='danger'>The reactive incendiary armor activates, but fails to send out flames as it is still recharging its flame jets!</span>"
+	emp_message = "<span class='warning'>The reactive incendiary armor's flame jets lock shut...</span>"
+	var/obj/item/clothing/gloves/flame_jets/gloves
+	var/broken = FALSE
+
+/obj/item/clothing/suit/armor/reactive/fire/Initialize()
+	. = ..()
+	gloves = new(src)
+	RegisterSignal(gloves, COMSIG_PARENT_PREQDELETED, .proc/armor_broken)
+
+/obj/item/clothing/suit/armor/reactive/fire/Destroy()
+	. = ..()
+	if(gloves)
+		QDEL_NULL(gloves)
+
+/obj/item/clothing/suit/armor/reactive/fire/proc/armor_broken(datum/source, force)
+	SIGNAL_HANDLER
 
-/obj/item/clothing/suit/armor/reactive/fire/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(!active)
+	if(isliving(loc))
+		var/mob/living/user = loc
+		to_chat(user, "<span class='danger'>[src]'s gloves are destroyed, breaking the reactive armor!</span>")
+		attack_self(user)
+
+/obj/item/clothing/suit/armor/reactive/fire/activate_effect(mob/living/carbon/human/owner)
+	if(broken)
+		to_chat(owner, "<span class='warning'>The flame jets are broken, preventing you from using this armor!</span>")
+		return FALSE
+	if(!ishuman(owner))
+		to_chat(owner, "<span class='warning'>The machine cannot fit the reactive incendiary gloves on you!</span>")
 		return FALSE
-	if(prob(hit_reaction_chance))
-		if(world.time < reactivearmor_cooldown)
-			owner.visible_message("<span class='danger'>The reactive incendiary armor on [owner] activates, but fails to send out flames as it is still recharging its flame jets!</span>")
-			return FALSE
-		owner.visible_message("<span class='danger'>[src] blocks [attack_text], sending out jets of flame!</span>")
-		playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
-		for(var/mob/living/carbon/C in range(6, owner))
-			if(C != owner)
-				C.adjust_fire_stacks(8)
-				C.IgniteMob()
-		owner.set_fire_stacks(-20)
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
+	if(!owner.equip_to_slot_if_possible(gloves, ITEM_SLOT_GLOVES, bypass_equip_delay_self = TRUE))
+		to_chat(owner, "<span class='warning'>Something is blocking the machine from equipping the reactive incendiary gloves on you!</span>")
+		return FALSE
+	owner.visible_message("<span class='notice'>[src] extends some metal gauntlets and they lock around [owner]'s hands!</span>")
```

Comment:
does reactive armor automatically deactivate when you take it off, or can you just have a friend activate the armor and then drop it for you so that you can wear it without the gloves?

Reply:
see below